docs/implementors-guide.md

```diff
@@ -0,0 +1,21 @@
+# Implementor's Guide
+
+## Purpose of the type and the provider fields in the application projection
+
+Service Binding specification mandates `type` field and recommends `provider` field in the binding Secret resource.  The provider field is recommended to support scenarios where there could be different providers for the same Provisioned Service type.  For example, if the type is `mysql`, the provider value could be `mariadb`, `oracle`, `bitnami`, `aws-rds`, etc.  When the application is reading the binding values, if necessary, the application could consider `type` and `provider` as a composite key to avoid ambiguity. It will be helpful if an application needs to choose a particular provider based on the deployment environment. 
+
+The language binding API can create two separate functions to retrieve bindings (with provider and without provider). For example, in Go:
+
+```
+GetBinding(_type string) (map[string]string, error)
+GetBindingWithProvider(_type, provider string) (map[string]string, error)
```

Comment:
I'd caution against attempting to return a single matching binding. `type` and `provider` are keys that can be used to query the set of bindings, that query may return zero, one or many matches.

Reply:
Thanks for the input!

I changed the API to look similar to that of Java.